pkg/client/apiutil/apimachinery.go

```diff
@@ -153,19 +154,6 @@ func RESTClientForGVK(gvk schema.GroupVersionKind, isUnstructured bool, baseConf
 	return rest.RESTClientFor(createRestConfig(gvk, isUnstructured, baseConfig, codecs))
 }
 
-// serializerWithDecodedGVK is a CodecFactory that overrides the DecoderToVersion of a WithoutConversionCodecFactory
-// in order to avoid clearing the GVK from the decoded object.
-//
-// See https://github.com/kubernetes/kubernetes/issues/80609.
```

Comment:
🤔 Without this, is the GVK preserved? The issue linked here is still open.

Reply:
The watch does preserve the GVK value, I added a test to prove this https://github.com/kubernetes-sigs/controller-runtime/pull/2147/commits/fa6aa5463b0de0a82bd435f1664a95052e67599a.